xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/scope/ScopeNotificationFilterExpressionGenerator.java

```diff
@@ -297,8 +295,7 @@ private boolean isEnabledScopeNotificationFilterPreference(NotificationFilterPre
     {
         // This optimization can only works on preferences stored by the user, that's why we add a condition
         // on the provider hint.
-        return nfp.isEnabled() && ScopeNotificationFilter.FILTER_NAME.equals(nfp.getFilterName())
-                && UserProfileNotificationFilterPreferenceProvider.HINT.equals(nfp.getProviderHint());
+        return nfp.isEnabled() && ScopeNotificationFilter.FILTER_NAME.equals(nfp.getFilterName());
```

Comment:
Hmmm I just realized that this method was having the following documentation:
> NOTE: this work only with user's preferences stocked in the database. It does not support the use-case of
     * the notifications macro, when the preferences are created on the fly based on the macro parameters!

I probably need to check it a bit more.

Reply:
I double checked it and the issue was about being able to check if the filter was stored or created on the fly: I decided to go for a check on the filter ID for this, since we use some internal prefixes for storing preferences.